src/storage-controller/src/lib.rs

```diff
@@ -2052,7 +2052,7 @@ where
             shard_name: id.to_string(),
             handle_purpose: format!("controller data for {}", id),
         };
-        let write = persist_client
+        let mut write = persist_client
```

Comment:
It it okay to move the write-handle open to _after_ the read handle open? It sounds like it's important for the write handle to start out newer than the read handle, semantically, and that would help make that more obvious. (And probably be enough to fix the bug all on its own, though I think the explicit call is not a bad idea...)

Reply:
yeah, let me go ahead and make this change

Reply:
done!